server/service/store/handler_store_data.go

```diff
@@ -0,0 +1,42 @@
+package store
+
+import (
+	"encoding/hex"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/gorilla/mux"
+	log "github.com/sirupsen/logrus"
+)
+
+func HandleStoreData(w http.ResponseWriter, r *http.Request) {
+	dataHashStr := mux.Vars(r)["key"]
+	_, err := hex.DecodeString(dataHashStr)
+	if err != nil {
+		log.Errorf("failed to decode dataHash: %s", err.Error())
+		http.Error(w, "failed to decode dataHash", http.StatusBadRequest)
+		return
+	}
+
+	body, err := io.ReadAll(r.Body)
+	defer r.Body.Close()
+	if err != nil {
+		log.Errorf("failed to read data: %v", err.Error())
+		http.Error(w, "failed to read data", http.StatusBadRequest)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/octet-stream")
+	w.WriteHeader(http.StatusCreated)
+	if _, err = w.Write(body); err != nil {
```

Comment:
What is the result of this response?
It looks like the client will receive a response with their file intact.

Reply:
It would seem that the client would simply receive an httpStatus code of 200 or 203 in response.

Reply:
Agree. No need to respond body back to client